pkg/handler/handler.go

```diff
@@ -354,6 +385,50 @@ func (m *Modifier) getPodSpecPatch(pod *corev1.Pod, roleName, audience string, r
 	return patch, changed
 }
 
+// buildPodPatchConfig reads configurations from multiples data sources and builds a merged podPatchConfig.
+// Data sources include: Cache, Config, and pod's annotations.
+// It calls Cache and Config to get the configurations of a service account
+//
+// Some mutation parameters can be overridden via pod or serviceaccount
+// annotations. The serviceaccount cache already parsed the serviceaccount
+// annotations and flags such that annotations take precedence.
+// audience:        serviceaccount annotation > flag
+// regionalSTS:     serviceaccount annotation > flag
+// tokenExpiration: pod annotation > serviceaccount annotation > flag
+func (m *Modifier) buildPodPatchConfig(pod *corev1.Pod) *podPatchConfig {
+	// Container credentials method takes precedence
+	containerCredentialsPatchConfig := m.Config.Get(pod.Namespace, pod.Spec.ServiceAccountName)
```

Comment:
So the behavior is: if the service account in present in the file, use the container cred env vars, otherwise use the web identity env vars.  The user will therefore populate the dynamic config file with a service account, and all new pod launches that reference that service account will get the container credential env vars. Then if the service account is removed from the file, launched pods will get the web identity env vars again.    As I mentioned in a previous comment, since the env var type is controlled by the SA's presence in `--watch-config-file` we should rename that flag or improve the description to show that it is tied to the feature.  

This reinforces the config file being explicitly tied to the container credential behavior, so if we think non-eks users might want to use the file with web identity behavior then perhaps we shouldn't go in that direction, and might want another configuration option which determines how the config file is used. That being said, its probably better to wait for someone to ask for that behavior before implementing.

Reply:
Done. Renamed newly introduced things to container-credentials